File: Top 100 Codes/03-Codes for Recursion/01-Power of a Number/PowerofaNumber.cpp
#include <bits/stdc++.h>
using namespace std;

// Recursive function to calculate power
int power(int base, int exponent) {
    if (exponent == 0) // Base condition
        return 1;
    return base * power(base, exponent - 1);
}

int main() {
    int base, exponent;
    
    // Taking user input
    cout << "Enter base: ";
    cin >> base;
    cout << "Enter exponent: ";
    cin >> exponent;

    // Display result
    cout << "Required Power is " << power(base, exponent) << endl;
}

File: Top 100 Codes/03-Codes for Recursion/01-Power of a Number/PowerofaNumber.py
def power(a, b):
    if b == 0:  # Base condition
        return 1
    return a * power(a, b - 1)

# Taking user input
a = int(input("Enter base: "))
b = int(input("Enter exponent: "))

# Displaying result
print(f"{a} to the power {b} is {power(a, b)}")
